src/io/file_descriptor.cr

```diff
@@ -5,9 +5,10 @@ class IO::FileDescriptor < IO
   include Crystal::System::FileDescriptor
   include IO::Buffered
 
-  # The raw file-descriptor. It is defined to be an `Int`, but its size is
-  # platform-specific.
-  def fd : Int
+  @volatile_fd : Atomic(Handle)
+
+  # Returns the raw file-descriptor handle. Its type is platform-specific.
+  def fd : Handle
```

Comment:
If `Handle` becomes `LibC::HANDLE`, wouldn't this type restriction need to be reverted back to `Int`?

Reply:
Yeah if we want to go that way, we might need to split between the internal and external representation of the handle. Let's take this step by step, though. `FileDescriptor:Handle` is supposed to be a generic name used for public APIs.

The exact type is platform-specific, but it should be a concrete type, not `Int`.
We could consider using the same type on all platforms in the future (`SizeT` probably). But I fear that would be too much of a breaking change for Unix systems right now.